General Updates and Refactors to Helm Docs (#21656)
gravitational/teleport: docs/pages/deploy-a-cluster/helm-deployments/aws.mdx

description: Install and configure an HA Teleport cluster using an AWS EKS cluster
---

> In this guide, we'll go through how to set up a High Availability Teleport cluster using Teleport Helm charts that runs on AWS EKS.

Reviewer: ```suggestion
In this guide, we'll use Teleport Helm charts to set up a High Availability Teleport cluster that runs on AWS EKS.
```

Alternate wording to make it clear that the Teleport cluster will run on EKS, not the helm charts.

### Determining an approach

> There are three supported options on AWS (choose one):

Author: I do get where you are coming from, maybe we can find a middle ground.  Would it be possible for choosing an approach to be summarized in a visual (a table of supported/unsupported features)? Does that seem too complicated?

(opinion) As a user, I'm use to having the recommended approach default, front and center.  It seems like we have a soft recommendation for cert-manager.  I wonder if given the number of drawbacks with AWS Certificate Manager, we shouldn't be more strongly encouraging users to pursue cert-manager if they are at all on the fence.  This probably explains why I'd more naturally lean towards promoting a preferred route while still providing optional paths and hiding unrelated information behind tabs/pages.

> There are three supported options on AWS (choose one):

Reviewer 2: I'm definitely all for reducing the number of options in order to simplify this page. @webvictim @hugoShaka What do you think of recommending cert-manager here?

> There are three supported options on AWS (choose one):

Reviewer 3: I don't think that's doable. We have 2 typical use cases:
- cert-manager + Teleport terminating TLS: the most versatile (better for us), more complex as it involves more maintenance on the user side. Also, might provide less security guarantees than ACM, requires to grant powerful DNS rights to a workload, not the best from a compliance POV.
- ACM + an Amazon-managed LB Terminating TLS: less portable (more work and complexity for us), worse UX because some features require TLS Termination, can't run TLS Multiplexing (yet). But it's simpler from a user pov, more secure, compliant, and likely already integrated in the customer stack.

Cert-manager-only is easier for us but worse for the users. A lot of users are using ACM because it's better for them and I don't think we can afford to remove it.

> There are three supported options on AWS (choose one):

Reviewer 2: @evanfreed Just checking to see if you have any thoughts on this thread.

> There are three supported options on AWS (choose one):

Reviewer 4: It's somewhat easier to use ACM now as Teleport 13 added support for multiplexing behind ALBs, which also opens up the ability to use a Kubernetes Ingress.

I'm working on a partial docs rewrite for this here: https://github.com/gravitational/teleport/pull/25815

> There are three supported options on AWS (choose one):

Author: > I don't think we can afford to remove it.

Plan wasn't to remove it, just to have different tabs for the differing cert options.  That way text is only displayed based on the users needs.

Sorry for the delay this fell off my radar. Given the discussion and my late response I'll just revert this part back to how it was and simplify the changes a bit.

  ```

</TabItem>

> <TabItem label="Your own TLS">

Reviewer 2: "Your own TLS" sounds a bit awkward to me. Can you own transport layer security?

Author: That's fair, I think most users reading this guide know they don't own the transport layer 😄 

```suggestion
<TabItem label="Your own TLS credentials">
```

</TabItem>
<TabItem label="Your own TLS credentials">

> ### Your own TLS

Author: ```suggestion
### Your own TLS credentials
```
